src/java.base/share/classes/java/util/FormatProcessor.java

```diff
@@ -0,0 +1,205 @@
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.template.ProcessorLinkage;
+import java.lang.template.StringProcessor;
+import java.lang.template.StringTemplate;
+import java.lang.template.ValidatingProcessor;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import jdk.internal.javac.PreviewFeature;
+
+/**
+ * This {@linkplain StringProcessor template processor} constructs a {@link String}
+ * result using {@link Formatter}. Unlike {@link Formatter}, {@link FormatProcessor} uses
+ * the value from the embedded expression that follows immediately after the
+ * <a href="../util/Formatter.html#syntax">format specifier</a>.
+ * StringTemplate expressions without a preceding specifier, use "%s" by
+ * default. Example:
+ * {@snippet :
+ * int x = 10;
+ * int y = 20;
+ * String result = FMT."%05d\{x} + %05d\{y} = %05d\{x + y}";
+ * }
+ * result is: <code>00010 + 00020 = 00030</code>
+ *
+ * @implNote When used in conjunction with a runtime instances of {@link
+ * StringTemplate} representing string templates this {@link StringProcessor}
+ * will use the format specifiers in the fragments and types of the values in
+ * the value list to produce a more performant formatter.
+ *
+ * @implSpec Since, values are found within the string template, argument indexing
+ * specifiers are unsupported.
+ *
+ * @since 21
+ */
+@PreviewFeature(feature=PreviewFeature.Feature.STRING_TEMPLATES)
+public final class FormatProcessor implements StringProcessor, ProcessorLinkage {
+    /**
+     * {@link Locale} used to format
+     */
+    private final Locale locale;
+
+    /**
+     * Constructor.
+     *
+     * @param locale  {@link Locale} used to format
+     */
+    public FormatProcessor(Locale locale) {
+        this.locale = locale;
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
Will clean up javadoc for FormatProcessor::process.

It is unfortunate. If you have any ideas will gladly take them on. On the other hand, not sure if the exception is any worse that a static initializer exception visually.

Reply:
Since FormatProcessor is scanning for format strings (in stringTempateFormat), it could check for the wrong number and misplaced format strings and throw.  It would make the exception more specific to the arguments to the FMT.process() and easier for the developer to correct.

Reply:
It is doing that now, but I can make the message more informative. The problem still exists that this is always done in the bootstrap method.